workspace/data-proxy/src/cli/utils/private-key.ts

```diff
@@ -15,6 +16,46 @@ async function readPrivateKeyFile(
 	return privateKeyFile;
 }
 
+export async function loadNetworkFromKeyFile(
+	privateKeyFilePath?: string,
+): Promise<Result<Environment, string>> {
+	const filePath = privateKeyFilePath ?? DEFAULT_PRIVATE_KEY_JSON_FILE_NAME;
+	const privateKeyFile = await readPrivateKeyFile(filePath);
+
+	if (privateKeyFile.isErr) {
+		return Result.err(
+			`Failed to read private key file ${filePath}: ${privateKeyFile.error}`,
+		);
+	}
+
+	const privateKeyFileObject = trySync(() =>
+		JSON.parse(privateKeyFile.value.toString()),
+	);
+
+	if (privateKeyFileObject.isErr) {
+		return Result.err(
+			`Failed to read private key file as JSON: ${privateKeyFileObject.error}`,
+		);
+	}
+
+	const parsedPrivateKeyFile = tryParseSync(
+		FileKeyPairSchema,
+		privateKeyFileObject.value,
+	);
+
+	if (parsedPrivateKeyFile.isErr) {
+		let resultError = "";
+
+		for (const error of parsedPrivateKeyFile.error) {
+			resultError += `${error.message} on config property "${error.path?.[0].key}" \n`;
+		}
+
+		return Result.err(`Failed to parse private key file: \n ${resultError}`);
+	}
+
+	return Result.ok(parsedPrivateKeyFile.value.network);
+}
+
 export async function loadPrivateKey(
```

Comment:
This is the part that is potentially refactorizable.
I did not do it because of time pressure and I didn't want to touch additional code.
Happy if anyone wants to do it.